src/Compilers/CSharp/Portable/Binder/Binder_Patterns.cs

```diff
@@ -1195,22 +1192,56 @@ void addSubpatternsForTuple(ImmutableArray<TypeWithAnnotations> elementTypes)
             DiagnosticBag diagnostics)
         {
             BoundExpression value = BindExpressionForPattern(inputType, node.Expression, hasErrors, diagnostics, out var constantValueOpt, out _);
+            BinaryOperatorKind operation = TokenKindToBinaryOperatorKind(node.OperatorToken.Kind());
+            BinaryOperatorKind opType = BinaryOperatorType(value.Type!.SpecialType);
+            switch (opType)
+            {
+                case BinaryOperatorKind.Float:
+                case BinaryOperatorKind.Double:
+                    if (!hasErrors && constantValueOpt != null && double.IsNaN(constantValueOpt.DoubleValue))
+                    {
+                        diagnostics.Add(ErrorCode.ERR_RelationalPatternWithNaN, node.Expression.Location);
+                        hasErrors = true;
+                    }
+                    break;
+                case 0:
```

Comment:
>0: [](start = 21, length = 2)

Why not use the name `BinaryOperatorKind.None` instead?

Reply:
No such thing.  I'll use `BinaryOperatorKind.Error`.

---
In reply to: [381002211](https://github.com/dotnet/roslyn/pull/41503#discussion_r381002211) [](ancestors = 381002211)